### PR TITLE
[SPARK-49139][SQL] Enable collations by default

### PR DESCRIPTION
**Pull request by uros-db**
### What changes were proposed in this pull request?
Set `COLLATION_ENABLED` SQL conf to `true` by default.

### Why are the changes needed?
Collations should now be enabled by default.

### Does this PR introduce _any_ user-facing change?
Yes, collations should now be enabled by default, without the need to set `spark.sql.collation.enabled` manually.

### How was this patch tested?
Existing tests.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Review by uros-db — COMMENTED** [review]
@cloud-fan all checks good - waiting for your approval to merge this to enable collations in Spark globally by default

**Comment by dbatomic** [comment]
Really happy to see this :)

**Review by dbatomic — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:768]: can we grep `spark.sql.collation.enable` in the codebase? IIRC we explicitly enable this conf in some examples of SQL function doc.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
May I ask why we need to enable this new feature (which is still documented as `Under development`) by default? We usually release new feature with disabled status at least one release, don't we? It looks normal and safer for me to have this `false` at Apache Spark 4.0.0 and enable it at 4.1.0. WDTY?

https://github.com/apache/spark/blob/7769ef11f3674f32f6e5405d13a52c328e8a730f/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L764

**Comment by cloud-fan** [comment]
@dongjoon-hyun String Collation is an opt-in featuer. You have to specify the collation to use this feature. We added a flag to avoid exposing a half-baked feature. Maybe we should just remove this conf as it's useless now, the feature is fully functional.

**Comment by dongjoon-hyun** [comment]
Got it. If that's the case, +1 for removal this conf. Thank you, @cloud-fan .

**Review by uros-db — COMMENTED** [review]
removed the config, @cloud-fan @dongjoon-hyun please re-review

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by dongjoon-hyun** [comment]
Thank you, @uros-db and @cloud-fan .

**Pull request closed without merging**
